### PR TITLE
Add celery task scheduler, book request, librarian issue, notifications and waiting list functionality.

### PR DESCRIPTION
**Pull request by shreyasrami**

### CODE REVIEW

File: libraryBackend/libraryApp/api.py [hunk 1]
```diff
@@ -160,206 +161,315 @@ def post(self, request, *args, **kwargs):
...
+            if book.available_quantity > 0:
+                copy = Copy.objects.filter(is_available=True, book=book)
+                copy = copy[0]
+                if request.user.is_student:
+                    student = Student.objects.get(email=request.user)
+                    WaitingList.objects.create(
+                        book=book,
+                        student=student,
+                        is_alerted=True,
+                        alerted_on=datetime.date.today(),
```
Comment (SakshiUppoor) [inline libraryBackend/libraryApp/api.py]: This line gives the following error `RuntimeWarning: DateTimeField Image.added_on received a naive datetime (2021-04-13 14:48:22) while time zone support is active.`

File: libraryBackend/libraryApp/api.py [hunk 2]
```diff
@@ -160,206 +161,315 @@ def post(self, request, *args, **kwargs):
...
+                    Notification.objects.create(
+                        nf_type="ALLOTED",
+                        user=teacher,
+                        notification="You are allocated the book "
+                        + book.name
+                        + ". Please collect it from the college library in 5 days",
+                    )
+                copy.is_available = False
+                copy.save()
+                return Response(issue.id)
```
Comment (SakshiUppoor) [inline libraryBackend/libraryApp/api.py]: Please return a more meaningful response.

File: libraryBackend/libraryApp/api.py [hunk 3]
```diff
@@ -160,206 +161,315 @@ def post(self, request, *args, **kwargs):
...
+                        "You are already there in the waiting list for this book"
+                    )
+                return Response(
+                    "Book is not available currently, you'll be notified once it's available"
+                )
+        else:
+            return Response("Book not found", status=HTTP_400_BAD_REQUEST)
+
+
+class IssuedViewSet(viewsets.ModelViewSet):
```
Comment (SakshiUppoor) [inline libraryBackend/libraryApp/api.py:474]: Students and teachers should not have update and delete permissions.

File: README.md [hunk 4]
```diff
@@ -40,6 +40,21 @@
...
+@"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "[System.Net.ServicePointManager]::SecurityProtocol = 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))" && SET "PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin"
+```
+2. To install RabbitMQ, run the following command from the command line or from PowerShell with administrative permission : 
+```bash
+chocolatey install rabbitmq
+```
+3. To start the rabbitmq server, open the RabbitMQ CLI and run the following command :
+```bash
+rabbitmq-server
+```
```
Comment (SakshiUppoor) [inline README.md:56]: Add instructions for the celery commands too.

**Review by SakshiUppoor — CHANGES_REQUESTED** [review]

**Comment by SakshiUppoor** [comment]
Also try making a file to run during setup which will automatically create the periodic task if it doesn't exist already.

**Comment by SakshiUppoor** [comment]
The code looks good to me. Just one doubt: If a student/teacher has issued a book, they shouldn't be able to issue it again if they haven't returned it, right?

**Comment by shreyasrami** [comment]
> The code looks good to me. Just one doubt: If a student/teacher has issued a book, they shouldn't be able to issue it again if they haven't returned it, right?

Fixed this.

**Pull request merged**